### PR TITLE
Add edit history to message sniping  and add command to view edit history of message

### PR DESCRIPTION
**Pull request by The-personified-devil**
And also fix fuzzywuzzy not being a dependency

### CODE REVIEW

File: src/cogs/rules.py [hunk 1]
```diff
@@ -29,24 +33,47 @@ async def rule(self, ctx, number):
         else:
             await ctx.send(f"**Rule {number}**:\n{self._rules[number]}")
 
+    def _change_msg(self, state, status):
+        user = str(state[1].author)
+        ts = datetime.fromtimestamp(state[0]).isoformat(" ", "seconds")
+        content = state[1].clean_content
+        return f"**{discord.utils.escape_markdown(discord.utils.escape_mentions(user))}** {status} on {ts} UTC:\n{content}\n"
+
```
Comment (Yuhanun) [inline src/cogs/rules.py]: what is `state`?

File: src/cogs/rules.py [hunk 2]
```diff
@@ -189,12 +216,29 @@ async def on_message_delete(self, message):
         channel = message.channel
 
         if channel not in self._deleted:
-            self._deleted[channel] = [message]
-            return
+            self._deleted[channel] = []
+
+        if message.id not in self._message_history:
+            self._deleted[channel].append([(time.time(), message)])
```
Comment (Yuhanun) [inline src/cogs/rules.py]: why not write a small data class for this?
```py
class Entry:
    time: float
    message: str
```

File: src/cogs/rules.py (hunk 1 above)
Comment (The-personified-devil) [reply src/cogs/rules.py]: It's a point in the history of a message

Comment (The-personified-devil) [reply src/cogs/rules.py]: Renamed it to `entry` to hopefully be less confusing

**Pull request merged**